### PR TITLE
Fixes Autoinjectors not going through armors.

### LINKED ISSUE #1048: Autoinjectors tomf*ckery
When trying to inject other people (sometimes even yourself), there will be cases of the game giving you an error message stating "You cannot directly fill this object" or some nonsense like that. Can become extremely disruptive to the point where it's impossible to administer medicine to someone.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
fixes #1048.

## Why It's Good For The Game
Resolving issues. Moving down on the road some combat logs that otherwise would be confusing (MP/MAA skillset blocking unwanted attempt log happening after the injection one.).

## Changelog
:cl:
fix: Fixes autoinjectors not working through armor and the such.
/:cl:

### CODE REVIEW

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -120,15 +120,8 @@
...
-
-	var/list/injected = list()
-	for(var/datum/reagent/R in reagents.reagent_list)
-		injected += R.name
-	var/contained = english_list(injected)
-	log_combat(user, A, "injected", src, "Reagents: [english_list(reagents.reagent_list)]")
-
-	if(ismob(A))
-		var/mob/M = A
+	if(isliving(A))
```
Comment (DominikPanic) [inline code/game/objects/items/reagent_containers/hypospray.dm]: You already check for living a few lines up.

Comment (Ghommie) [reply code/game/objects/items/reagent_containers/hypospray.dm]: That actually checks all other atoms but mobs.

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 2]
```diff
@@ -138,6 +133,17 @@
...
+			return FALSE
+
+	var/list/injected = list()
+	for(var/datum/reagent/R in reagents.reagent_list)
+		injected += R.name
+	var/contained = english_list(injected)
+	log_combat(user, A, "injected", src, "Reagents: [english_list(reagents.reagent_list)]")
+
+	if(ismob(A))
+		var/mob/living/M = A
```
Comment (DominikPanic) [inline code/game/objects/items/reagent_containers/hypospray.dm]: You're checking for ismob but typecasting a living.

Comment (Ghommie) [reply code/game/objects/items/reagent_containers/hypospray.dm]: Whops, thanks for spotting.

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 3]
```diff
@@ -138,6 +133,17 @@
 			playsound(user.loc, 'sound/weapons/thudswoosh.ogg', 25, 1, 7)
 			return FALSE
 
+		if(!M.can_inject(user, TRUE, user.zone_selected, TRUE))
```
Comment (DominikPanic) [inline code/game/objects/items/reagent_containers/hypospray.dm]: Why the code duplication here? (same check a few lines up)

Comment (Ghommie) [reply code/game/objects/items/reagent_containers/hypospray.dm]: Distracted there, now fixed.

**Pull request merged**